thunder/torch/__init__.py

```diff
@@ -3489,6 +3489,21 @@ def embedding_backward(grad, indices, num_weights, padding_idx, scale_grad_by_fr
     return result
 
 
+@torchsymbol(torch.nn.functional.one_hot, id="torch.nn.functional.one_hot", is_method=False)
+def one_hot(a: TensorProxy, /, num_classes: int) -> TensorProxy:  # Should we use TensorProxy for a?
+    if num_classes == -1:
+        # TODO: implement when we're ready to support auto inference using `.item()`
+        utils.check(False, lambda: "num_classes=-1 not supported yet.", exception_type=NotImplementedError)
+    # TODO: should we implement this check? I suspect this might hinder performance, also we do not support .any)
+    #  utils.check(a.any() >= 0, lambda f"input tensor should have non-negative values", exception_type=ValueError)
+
+    canvas = zeros(*a.shape, num_classes, dtype=dtypes.int64)  # this is equivalent to the use of .long in torch
```

Comment:
The test failures are because this tensor is always allocated on the cpu. 

The best way to fix this (which doesn't need to happen in this PR) is to extend the "like" functions, like `zeros_like`

https://github.com/shaharelys/lightning-thunder/blob/9ff2c62491e52d580d8acfead46e15b40773fa1e/thunder/torch/__init__.py#L526

to accept a `shape` parameter that can override the shape of the example tensor. Then the line could be:

```
result_shape: tuple[int, ...] = a.shape + (num_classes,)
canvas = zeros_like(a, shape=result_shape)
```

and developers wouldn't need to worry about setting additional metadata.

For now, the simplest fix is to just set the device:

```
canvas = zeros(*a.shape, num_classes, device=a.device, dtype=dtypes.int64)
```

fyi @nikitaved to consider the extension of the "like" operations

Reply:
We have `zeros_like`

Reply:
We have `zeros_like`, but I don't think it has a shape parameter to override the example tensor's shape?

Reply:
Was not sure about this one